std/algorithm/sorting.d

```diff
@@ -3100,7 +3102,8 @@ Stable topN has not been implemented yet.
 auto topN(alias less = "a < b",
         SwapStrategy ss = SwapStrategy.unstable,
         Range)(Range r, size_t nth)
-if (isRandomAccessRange!(Range) && hasLength!Range && hasSlicing!Range)
+if (isRandomAccessRange!(Range) && hasLength!Range &&
+    hasSlicing!Range && hasAssignableElements!Range)
```

Comment:
Shouldn't this be hasSwappableElements?

Reply:
No it calls `pivotPartition` which calls `moveAt` and assigns to the element:

```d
auto save = r.moveAt(hi);
...
r[lo] = save;
```

`hasSwappableElements` just requires that `swap(a, b)` works.